src/sqlfluff/cli/commands.py

```diff
@@ -575,6 +576,28 @@ def lint(
         sys.exit(0)
 
 
+def _mark_failed_files_unfixable(lint_result: LintingResult) -> int:
+    """Discard lint fixes for files with templating or parse errors.
+
+    Returns 1 if there are any files with templating or parse errors after
+    filtering, else 0. (Intended as a process exit code.)
+    """
+    total_errors, num_filtered_errors = lint_result.mark_failed_files_unfixable()
+    if total_errors:
+        click.echo(
+            colorize(f"  [{total_errors} templating/parsing errors found]", Color.red)
+        )
+        if 0 < num_filtered_errors < total_errors:
```

Comment:
>What if there are no filtered errors at all but there are TMP/PRS errors?
`num_filtered_errors` is a variable that counts the number of filtered TMP and PRS errors, so I don't understand the question.

>Should it be 0 <= num_filtered_errors < ...? At which point do you even need this and should it just be num_filtered_errors < total_errors?
I originally didn't have the `0 <` part. I added it because I was thinking, if *all* the errors are filtered, why print an "error" message saying "0 errors after filtering", because that's _good_, right?

Reply:
I removed the `0 < ` from the `if`, so now the `no cover` shouldn't be needed, because existing tests will now hit this code. Instead, I changed it so the color of the output is red if there are still errors after filtering, green if there are none.